### PR TITLE
Extended filtering scenarios

### PR DESCRIPTION
**Pull request by MattCzerner**

### CODE REVIEW

File: scenario.scala [hunk 1]
```diff
@@ -98,6 +112,12 @@ object Walk {
     .pause(100.milliseconds)
     .exec(categoryListFilterPrice)
     .pause(100.milliseconds)
+    .exec(categoryListFilterPriceBrand)
+    .pause(100.milliseconds)
```
Comment (boris-brtan) [inline scenario.scala]: is it important to have 100ms delay, why not just 10ms or why not 1000ms ?

File: scenario.scala [hunk 2]
```diff
@@ -81,6 +89,12 @@ object Walk {
     .pause(100.milliseconds)
     .exec(categoryListFilterPrice)
     .pause(100.milliseconds)
+    .exec(categoryListFilterPriceBrand)
```
Comment (boris-brtan) [inline scenario.scala]: seems like the whole block misses 4 more whitespaces

Comment (boris-brtan) [reply scenario.scala]: Where is the search property defined?
 Is it still used ? 
`val searchOption = System.getProperty("search")`
Is the whole block `if (searchOption == null) {` still needed ?

File: scenario.scala [hunk 3]
```diff
@@ -38,14 +38,22 @@ object Walk {
...
     .get("/electronics/?product_filter_form[minimalPrice]=0&product_filter_form[maximalPrice]=20000")
     .headers(headers_ajax)
 
-  val categoryListFilterPriceBrand = http("Category list filtering")
+  val categoryListFilterPriceBrand = http("Category list filtering by price with brand selected")
     .get("/electronics/?product_filter_form[minimalPrice]=0&product_filter_form[maximalPrice]=20000&product_filter_form[brands][]=4")
     .headers(headers_ajax)
 
+  val categoryListFilterParameters = http("Category list filtering by parameters")
+    .get("/electronics/?=product_filter_form[parameters][3][]=6&product_filter_form[parameters][4][]=8&product_filter_form[parameters][4][]=12")
```
Comment (boris-brtan) [inline scenario.scala]:
```suggestion
    .get("/electronics/?product_filter_form[parameters][3][]=6&product_filter_form[parameters][4][]=8")
```

Comment (boris-brtan) [reply scenario.scala]: also looks like there is no USB parameter option 12 and there seems to be a bug shopsys/shopsys#929 when one parameter doesn't exist, so the existing one is not checked

Comment (boris-brtan) [reply scenario.scala]: maybe pick some more complex combination

File: scenario.scala [hunk 4]
```diff
@@ -38,14 +38,22 @@ object Walk {
...
+    .headers(headers_ajax)
+
+  val categoryListFilterAllFilters = http("Category list filtering by all possible filters")
+    .get("/electronics/?product_filter_form[minimalPrice]=200&product_filter_form[maximalPrice]=30000&product_filter_form[inStock]=1&product_filter_form[brands][]=5&product_filter_form[parameters][5][]=8&product_filter_form[parameters][1][]=2&product_filter_form[parameters][1][]=14&product_filter_form[parameters][4][]=8")
```
Comment (boris-brtan) [inline scenario.scala]: can you pick smaller range for prices? 200 and 30000 are now in last demo data build over the real MIN and MAX from electronics.

Comment (boris-brtan) [reply scenario.scala]: actually there is available also options in parameters filter
Resolution (id 3 value 6)
Technology  (id 2 value 4)
and Flags filter is not used

**Review by boris-brtan — CHANGES_REQUESTED** [review]
Hi,
THX for the improvement,
i just have few questions and suggestions.

File: scenario.scala (hunk 1 above)
Comment (MattCzerner) [reply scenario.scala]: I dont care. I dont see reason why I should change this established delay in this PR

File: scenario.scala (hunk 2 above)
Comment (MattCzerner) [reply scenario.scala]: I don't understand, I think that it is aligned well. Or not?

Also, I don't understand the `System.getProperty("search")`, so I would guess that yeah, it is still needed.

Comment (boris-brtan) [reply scenario.scala]: it seems that search is available also for 1 and 32 users since 10.8.2018

File: scenario.scala (hunk 1 above)
Comment (boris-brtan) [reply scenario.scala]: OK.

File: scenario.scala (hunk 4 above)
Comment (boris-brtan) [reply scenario.scala]: OK

Comment (boris-brtan) [reply scenario.scala]: RESOLVED

File: scenario.scala (hunk 1 above)
Comment (boris-brtan) [reply scenario.scala]: RESOLVED

File: scenario.scala (hunk 2 above)
Comment (boris-brtan) [reply scenario.scala]: for the alignment
image

Comment (boris-brtan) [reply scenario.scala]: RESOLVED

**Pull request merged**